generators/generator-botbuilder/generators/app/templates/echo/bot.ts

```diff
@@ -1,41 +1,48 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-const { ActivityTypes } = require('botbuilder');
+import { ActivityTypes, ConversationState, StatePropertyAccessor, TurnContext } from 'botbuilder';
+
 
 // Turn counter property
 const TURN_COUNTER_PROPERTY = 'turnCounterProperty';
 
 export class MyBot {
-    public countProperty;
-    public conversationState;
-  /**
-   *
-   * @param {ConversationState} conversation state object
-   */
-  constructor(conversationState) {
-    // Creates a new state accessor property.
-    // See https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors.
-    this.countProperty = conversationState.createProperty(TURN_COUNTER_PROPERTY);
-    this.conversationState = conversationState;
-  }
-  /**
-   *
-   * @param {TurnContext} on turn context object.
-   */
-  async onTurn(turnContext) {
-    // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
-    if (turnContext.activity.type === ActivityTypes.Message) {
-      // read from state.
-      let count = await this.countProperty.get(turnContext);
-      count = count === undefined ? 1 : ++count;
-      await turnContext.sendActivity(`${count}: You said "${turnContext.activity.text}"`);
-      // increment and set turn counter.
-      await this.countProperty.set(turnContext, count);
-    } else {
-      await turnContext.sendActivity(`[${turnContext.activity.type} event detected]`);
+    
+    private readonly countAccessor: StatePropertyAccessor<number>;
+    private readonly conversationState: ConversationState;
+
+    /**
+     * 
+     * @param {ConversationState} conversation state object
```

Comment:
The syntax highlighting on this line seems a bit odd; we might want to double check this docstring.

Reply:
this highlighting is the same as other docstrings in this file

Reply:
I looked into this a litle more, the docstrings here are for JS, not TS, we use [JavaDoc(?)](http://typedoc.org/guides/doccomments/) syntax, here's an example for [CardFactory.audioCard()](https://github.com/Microsoft/botbuilder-js/blob/master/libraries/botbuilder-core/src/cardFactory.ts#L102-L116):

```ts
/**
 * Returns an attachment for an audio card.
  * @param title The cards title.
  * @param media Media URL's for the card.
  * @param buttons (Optional) set of buttons to include on the card.
  * @param other (Optional) additional properties to include on the card.
  */
public static audioCard(
  title: string,
  media: (MediaUrl | string)[],
  buttons?: (CardAction | string)[],
  other?: Partial<AudioCard>
): Attachment {
  return mediaCard(CardFactory.contentTypes.audioCard, title, media, buttons, other);
}
```

That said, TS does support [JSDoc](https://github.com/Microsoft/TypeScript/wiki/JSDoc-support-in-JavaScript), so if the current syntax is kept, the first token after the `@param {ConversationState} ` should be the parameter name, i.e.:

```ts
/**
 *
 * @param {ConversationState} conversationState <Brief conversation state description>
 */
```